rgbd/test/shm_server_gtest.cpp

```diff
@@ -0,0 +1,106 @@
+#include "test_utils.h"
+
+#include <gtest/gtest.h>
+
+#include <boost/interprocess/shared_memory_object.hpp>
+
+#include <rgbd/image.h>
+#include <rgbd/server_shm.h>
+
+#include <ros/init.h>
+#include <ros/node_handle.h>
+#include <ros/subscriber.h>
+
+#include <std_msgs/String.h>
+
+#include <memory>
+#include <thread>
+
+
+namespace ipc = boost::interprocess;
+
+
+class SHMServer : public testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        image = rgbd::generateRandomImage();
+        server.initialize(test_server_name);
+    }
+
+    const std::string test_server_name = "/test_ns/rgbd";
+    const std::string test_server_name_shm = "-test_ns-rgbd";
+
+    rgbd::Image image;
+    rgbd::ServerSHM server;
+};
+
+class SHMServerHostame : public SHMServer
+{
+protected:
+    SHMServerHostame() : SHMServer(), correct_hostname(true)
+    {
+    }
+
+    void SetUp() override
+    {
+        sub = nh.subscribe<std_msgs::String>(test_server_name_hosts, 10, &SHMServerHostame::hostsCallback, this);
+    }
+
+    void hostsCallback(const std_msgs::String::ConstPtr& msg)
+    {
+        if (msg->data != test_hostname)
+            correct_hostname = false;
+    }
+
+    const std::string test_server_name_hosts = "/test_ns/rgbd/hosts";
+    const std::string test_hostname = "test_hostname";
+
+    ros::NodeHandle nh;
+    ros::Subscriber sub;
+    bool correct_hostname;
+};
+
+TEST_F(SHMServer, Initialize)
```

Comment:
Maybe change the test name to something like "Initialize when SHM already exists" or whatever you call it. I think that is whats being tested here.

Reply:
No, here is check the SHM is created. This is checked by trying to only open it. So the test can NOTcreate it.